apps/web/src/app/[locale]/challenge/_components/comments/comment.action.ts

```diff
@@ -47,9 +47,8 @@ export async function replyComment(comment: CommentToCreate, parentId: number) {
 
   const { rootId, ...commentToCreate } = {
     ...comment,
-    ...(comment.rootType === 'CHALLENGE'
-      ? { rootChallengeId: comment.rootId }
-      : { rootSolutionId: comment.rootId }),
+    rootChallengeId: comment.rootChallengeId,
```

Comment:
this part: why does this need to change?

Reply:
I thought this made things a bit more clear and explicit.

In the base case, normal comments work with `rootId = rootChallengeId`

Solution comments is kind of a special case
Where `rootId = rootSolutionId` and challengeId is lost in this situation.

I felt this change made that more explicit.

Reply:
the type is the discriminator here

Reply:
Reverted these changes and acting on your feedback.